src/test/java/com/maddisportfolio/user/UserServiceTest.java

```diff
@@ -0,0 +1,427 @@
+
+        //Act
+        List<User> returnedList = serviceUnderTest.searchUsers(firstName, lastName, loggedInUserEmailAddress);
+
+        //Assert
+        Assert.assertEquals(1, returnedList.size());
+        Assert.assertTrue(kupo.isHasReceivedRequestFromSearchedUser());
+        Assert.assertEquals(kupo, returnedList.get(0));
+    }
+
+    @Test
+    public void testSearchUsersIsFriendsByRecipient(){
+        //Arrange
+        User navi = new User();
+        User kupo = new User();
+        long id = 10;
+        kupo.setId(id);
+        String firstName = "kupo";
+        kupo.setFirstName(firstName);
+        String lastName = "mcconville";
+        kupo.setLastName(lastName);
+        String loggedInUserEmailAddress = "navi@cat.com";
+        navi.setEmailAddress(loggedInUserEmailAddress);
+        FriendRequest friendRequest = new FriendRequest();
+        List<FriendRequest> correspondingFriendRequests = new ArrayList<>();
+        correspondingFriendRequests.add(friendRequest);
+        friendRequest.setFriendRequestStatus(FriendRequestStatus.ACCEPTED);
+        friendRequest.setRecipient(navi);
+        friendRequest.setSender(kupo);
+        List<User> searchResults = new ArrayList<>();
+        searchResults.add(kupo);
+
+        Mockito.when(userDao.findOneByEmailAddressIgnoreCase(loggedInUserEmailAddress)).thenReturn(navi);
+        Mockito.when(userDao.findByFirstNameIgnoreCaseContainingAndLastNameIgnoreCaseContaining(firstName, lastName)).thenReturn(searchResults);
+        Mockito.when(friendRequestDao.findAllByRecipientAndFriendRequestStatus(navi, FriendRequestStatus.ACCEPTED)).thenReturn(correspondingFriendRequests);
+
+        //Act
+        List<User> returnedList = serviceUnderTest.searchUsers(firstName, lastName, loggedInUserEmailAddress);
+
+        //Assert
+        Assert.assertEquals(1, returnedList.size());
+        Assert.assertTrue(navi.isHasReceivedRequestFromSearchedUser());
+        Assert.assertEquals(kupo, returnedList.get(0));
+        Assert.assertTrue(kupo.isFriendsWithLoggedInUser());
+    }
+
+    @Test
+    public void testUpdateUserProfile(){
+        User user = new User();
+        String firstName = "Kupo";
+        user.setFirstName(firstName);
+        String lastName = "McConville";
+        user.setLastName(lastName);
+        String location = "Colorado Springs";
+        user.setLocation(location);
+        String timeZone = "MST";
+        user.setTimezone(timeZone);
+        String loggedInUserEmailAddress = "kupo@cat.com";
+        user.setEmailAddress(loggedInUserEmailAddress);
+
+        Mockito.when(userDao.findOneByEmailAddressIgnoreCase(loggedInUserEmailAddress)).thenReturn(user);
+
+        serviceUnderTest.updateUserProfile(firstName, lastName, location, loggedInUserEmailAddress, timeZone);
+
+        Mockito.verify(userDao).save(user);
```

Comment:
So here, you'd want to not set the first name, last name, etc on the user object, but you want to return a blank user object from your mock DAO call and then use an argument captor to assert that all the fields were filled out on it when it was saved